CMS: Adds new article on `umbraco-extension` (#7076)
umbraco/UmbracoDocs: 16/umbraco-cms/customizing/development-flow/umbraco-extension-template.md

- `MyExtension.csproj`: The project file for the extension.
- `Constants.cs`: A file containing constants for the extension.
- `Client`: A folder containing the source code for the extension, a package.json file, a tsconfig.json file, and the vite.config.ts configuration file.
- `README.md`: A readme file with instructions on how to build and run the extension.

The `-ex` flag indicates that you want to include examples of how to use the extension. This flag is optional, but it is recommended to include it if you are new to building extensions for Umbraco. It will additionally give you:

- `Composers`: A folder containing an example composer that registers a custom Swagger API.
- `Controllers`: A folder containing an example API controller for a dashboard.
- `Client/src/api`: A folder containing an example API client that calls the API controller.
- `Client/src/dashboards`: A folder containing an example dashboard Web Component that uses the API client.

You should be able to see the dashboard in the main Content section of the Backoffice.

### Include the Extension in your Umbraco Project

> To include the extension in your Umbraco project, you need to add a reference to the extension project in your Umbraco project. You can do this by right-clicking on the **Dependencies** node in your Umbraco project and selecting **Add Reference**. Then select the `MyExtension` project and click **OK**.

Reviewer: ```suggestion
To include the extension in your Umbraco project, you need to add a reference to the extension project in your Umbraco project. 

#### Option 1: Via Visual Studio

1. Right-click the **Dependencies** node in the Umbraco project.
2. Select **Add Reference**.
3.  Choose the `MyExtension` project.
4. Click **OK**.
```

> You can also add the reference manually by running the following command in the root folder of your Umbraco project:

Reviewer: ```suggestion
#### Option 2: Via CLI
Run the following command in the root folder of your Umbraco project:
```

```bash
dotnet add reference ../MyExtension/MyExtension.csproj
```

This command adds a reference to the `MyExtension` project in your Umbraco project. You can then build your Umbraco project and see the extension in action.

## Build and Run the Extension

> To build and run the extension, you need to install the dependencies and start the Vite development server. You can do this by running the following commands in the `Client` folder of your extension project:

Reviewer: ```suggestion
To build and run the extension, install the dependencies and start the Vite development server. To do this, run the following commands in the `Client` folder of your extension project:
```

```bash
npm install
npm run build
```

{% hint style="info" %}

> The project will also automatically build when you run the Umbraco project. You can also run the Vite development server in watch mode by running the following command:

Reviewer: ```suggestion
The project also builds automatically when running the Umbraco project. To start the Vite development server in watch mode, run the following command:
```

```bash
npm run watch
```
{% endhint %}

This command compiles the TypeScript files and copies them over to the output folder in `wwwroot`. You can then run your Umbraco project and see the extension in action.

## Publish the Project

The output files should automatically be copied to the `wwwroot` folder of your Umbraco project. They should also be included in the publish process when you publish your Umbraco project. You can publish your Umbraco project using the following command:

```bash

> dotnet publish -c Release

Reviewer 2: ```suggestion
dotnet publish --configuration Release
```
I had to look up what `-c` meant, (and then I was like "Doh!" 🤦)
Suggesting to use the verbose argument name (for less cognitive load).

```

## Publish as a Package

To publish your extension as a package, you need to create a NuGet package. You can do this by running the following command in the root folder of your extension project:

```bash

> dotnet pack -c Release

Reviewer 2: ```suggestion
dotnet pack --configuration Release
```
